app/controllers/organizations_controller.rb

```diff
@@ -51,16 +51,12 @@ def update
   def destroy
     organization = Organization.find_by(id: params[:id])
     authorize organization
-    if organization.destroy
-      current_user.touch(:organization_info_updated_at)
-      EdgeCache::BustUser.call(current_user)
-      flash[:settings_notice] = "Your organization: \"#{organization.name}\" was successfully deleted."
-      redirect_to user_settings_path(:organization)
-    else
-      flash[:settings_notice] = "#{organization.errors.full_messages.to_sentence}.
-        Please email #{SiteConfig.email_addresses[:contact]} for assistance."
-      redirect_to user_settings_path(:organization, id: organization.id)
-    end
+
+    Organizations::DeleteWorker.perform_async(organization.id, current_user.id)
+    flash[:settings_notice] =
+      "Your organization: \"#{organization.name}\" deletion is scheduled. You'll be notified when it's deleted."
```

Comment:
**suggestion (non-blocking):** Ruby has one more string literal that allows interpolation and doesn't need `"` escaped.

```suggestion
       %{Your organization: "#{organization.name}" deletion is scheduled. You'll be notified when it's deleted.}
```

Reply:
TIL!